name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/heatmap/ColorSchema.java

```diff
@@ -24,46 +26,92 @@ private ColorSchema(String label)
     {
-        return label;
+        return switch (this)
+        {
+            case GREEN_YELLOW_RED -> performance -> {
+                // Normalize performance between -0.07 and +0.07 and map it to a
+                // hue between 0 (red) and 120 (green)
+                double p = normalizePerformance(performance);
+
+                // 0 = red, 60 = yellow, 120 = green
+                float hue = (float) p * 120f;
+
+                return resourceManager.createColor(new RGB(hue, 0.9f, 1f));
+            };
+
+            case GREEN_WHITE_RED -> performance -> {
+                // Performance is normalized and interpolated between white
+                // (neutral) and green (positive) or red (negative)
+                double p = Math.min(MAX_PERFORMANCE, Math.abs(performance)) / MAX_PERFORMANCE;
+
+                RGB color = performance > 0f
+                                ? Colors.interpolate(Colors.WHITE.getRGB(), Colors.HEATMAP_DARK_GREEN.getRGB(), (float) p)
+                                : Colors.interpolate(Colors.WHITE.getRGB(), Colors.RED.getRGB(), (float) p);
+
+                return resourceManager.createColor(color);
+            };
+
+            case GREEN_RED_GREY -> performance -> {
+                // Performance interpolates between grey (neutral), green
+                // (positive), or red (negative)
+                double p = normalizePerformance(performance);
+
+                RGB color = performance > 0 ? Colors.interpolate(Colors.GRAY.getRGB(), Colors.HEATMAP_DARK_GREEN.getRGB(), (float) p)
+                                : Colors.interpolate(Colors.GRAY.getRGB(), Colors.RED.getRGB(), (float) p);
+
+                return resourceManager.createColor(color);
+            };
+
+            case BLUE_ORANGE_GREY -> performance -> {
+                // Performance interpolates between grey (neutral), blue
+                // (stable), or orange (volatile)
+                double p = normalizePerformance(performance);
+
+                RGB color = performance > 0
+                                ? Colors.interpolate(Colors.GRAY.getRGB(), Colors.BLUE.getRGB(), (float) p)
+                                : Colors.interpolate(Colors.GRAY.getRGB(), Colors.HEATMAP_ORANGE.getRGB(), (float) p);
+
+                return resourceManager.createColor(color);
+            };
+
+            case BLACK_YELLOW_WHITE -> performance -> {
+                // Performance interpolates between white (neutral), yellow
+                // (moderate), or black (extreme)
+                double p = normalizePerformance(performance);
+
+                RGB color = performance > 0.05
+                                ? Colors.interpolate(Colors.YELLOW.getRGB(), Colors.BLACK.getRGB(),
+                                                (float) ((p - 0.05) / 0.95))
+                                : Colors.interpolate(Colors.WHITE.getRGB(), Colors.YELLOW.getRGB(), (float) p);
```

Comment:
I have to admit, I do not understand the math. The 0.05 are deducted from the normalized 0 -> 1 range. As we normalize between -0.07 and +0.07, there are 0.14 on the range and 5% of that (the deduction) is 0.07.

Plus values that are above 0.07 will be black anyway (interpolation factor 1). As only performance > 0.05 is going into the black branch, the lower bound is 0.8571428571. The ```(p - 0.05) / 0.95)``` will rescale this to 0.8496240602. What difference does it make?


BTW, can you share where you picked up the black-yellow color palette? I looked for materials in the web, but was [not](https://www.datacamp.com/tutorial/seaborn-heatmaps) [really](https://vwo.com/blog/heatmap-colors/) [successful](https://www.atlassian.com/data/charts/how-to-choose-colors-data-visualization) in finding examples of black-yellow.